### ISSUE TITLE
dontStrip has no effect on Haskell packages

### ISSUE THREAD

**Issue by nh2** [NixOS/nixpkgs#43506]
## Issue description

Passing `dontStrip = true;` to a Haskell package has no effect, the resulting binary is `stripped`.

### Steps to reproduce

```
% file $(nix-build --expr 'with import <nixpkgs> {}; haskell.lib.overrideCabal haskellPackages.hello (old: { dontStrip = true; })')/bin/hello
/nix/store/qqf2c5zq5dqs69vhf7lal08kvb420nwz-hello-1.0.0.2/bin/hello: ELF 64-bit LSB executable, x86-64, version 1 (SYSV), dynamically linked, interpreter /nix/store/2kcrj1ksd2a14bm5sky182fv2xwfhfap-glibc-2.26-131/lib/ld-linux-x86-64.so.2, for GNU/Linux 2.6.32, stripped
```

## Technical details

```
% nix-shell -p nix-info --run 'nix-info -m'
 - system: `"x86_64-linux"`
 - host os: `Linux 4.13.0-43-generic, Ubuntu, 16.04.4 LTS (Xenial Xerus)`
 - multi-user?: `no`
 - sandbox: `no`
 - version: `nix-env (Nix) 2.0.4`
 - channels(niklas): `"nixpkgs-18.03.132865.411cc559c05, unstable-18.09pre145679.dae9cf6106d, nixos-17.03-17.03.873.0c041520c3"`
 - nixpkgs: `/home/niklas/.nix-defexpr/channels/nixpkgs`
```

CC @peti

**Comment by nh2** [comment]
I have checked that at the time `postInstall` runs,

```
          ${pkgs.file}/bin/file dist/build/hello/hello
```

shows as `not stripped` but

```
          ${pkgs.file}/bin/file $out/bin/hello
```

shows as `stripped`.

**Comment by nh2** [comment]
`nix-build --expr 'with import <nixpkgs> {}; haskell.lib.overrideCabal haskellPackages.hello (old: { dontStrip = true; postInstall = "${pkgs.file}/bin/file dist/build/hello/hello $out/bin/hello"; })'`

```
...
dist/build/hello/hello:                                              ELF 64-bit LSB executable, x86-64, version 1 (SYSV), dynamically linked, interpreter /nix/store/2kcrj1ksd2a14bm5sky182fv2xwfhfap-glibc-2.26-131/lib/ld-linux-x86-64.so.2, for GNU/Linux 2.6.32, with debug_info, not stripped
/nix/store/4anj1rmy0y4g9fkmcx0kzhf9jjmqwxvk-hello-1.0.0.2/bin/hello: ELF 64-bit LSB executable, x86-64, version 1 (SYSV), dynamically linked, interpreter /nix/store/2kcrj1ksd2a14bm5sky182fv2xwfhfap-glibc-2.26-131/lib/ld-linux-x86-64.so.2, for GNU/Linux 2.6.32, stripped
...
```

**Comment by nh2** [comment]
The solution seems to be to pass [`--disable-executable-stripping`](https://cabal.readthedocs.io/en/latest/nix-local-build.html#cfg-field-executable-stripping) in `configureFlags` when `dontStrip` is given.

**Comment by nh2** [comment]
PR with the fix at #43508
